tests/drivers/i2s/i2s_api/src/common.c

```diff
@@ -158,24 +156,24 @@ int configure_stream(const struct device *dev_i2s, enum i2s_dir dir)
 	int ret;
 	struct i2s_config i2s_cfg;
 
-	i2s_cfg.word_size = 16U;
+	i2s_cfg.word_size = (uint8_t)CONFIG_I2S_TEST_WORD_SIZE;
 	i2s_cfg.channels = 2U;
-	i2s_cfg.format = I2S_FMT_DATA_FORMAT_I2S;
+	i2s_cfg.format = (uint8_t)CONFIG_I2S_TEST_DATA_FORMAT;
```

Comment:
Why are these options configured through Kconfig? Changing them requires separate builds. Couldn't they be just parameters passed to this function?

Reply:
This does not justify the use of Kconfig in my opinion. Such approach is not scalable. Just see that you had to add four test scenarios (each one needs to be built separately) to cover only a few possible configurations.
The test could check if particular configurations are supported on a given device and skip them if not.

Reply:
I doubt that there is a straight forward way to check if word size configuration is supported for given device.
Moreover with the word size change the and without the K-config how do i address the comment regarding the
`static void fill_buf(**intt32_t** *tx_block, int att)`

The tx_block will need to change during compilation.

Furthermore making the word size runtime configurable will require passing parameters 
to Z_TEST before method

`ZTEST_SUITE(i2s_loopback, NULL, setup, before, NULL, NULL);`

How do i do that ?

Reply:
@anangl

Reply:
> I doubt that there is a straight forward way to check if word size configuration is supported for given device.

Checking the value returned by `i2s_configure()` should be straightforward enough.

> Moreover with the word size change the and without the K-config how do i address the comment regarding the `static void fill_buf(**intt32_t** *tx_block, int att)`
> 
> The tx_block will need to change during compilation.

You can typecast the pointer according to the configured word size.

> Furthermore making the word size runtime configurable will require passing parameters to Z_TEST before method

The I2S device can be configured in test functions, it does not have to be done in `before()`.

Reply:
Yes that is true but this will require modifying all test cases files, because configure_stream(...) is used everywhere. I would like to avoid this.

Reply:
@anangl lets have a meeting and discuss the best solution